cdap-ui/app/cdap/main.js

```diff
@@ -358,22 +357,35 @@ class CDAP extends Component {
               </div>
             </If>
             <If condition={!this.state.isNamespaceFetchInFlight}>
-              <If
-                condition={this.state.pageLevelError && this.state.pageLevelError.errorCode === 404}
-              >
+              <If condition={isNamespaceNotFound}>
                 <Page404 message={this.state.pageLevelError.message} />
               </If>
+              {/** We show authorization failure message when the specific namespace API returns 403 */}
+              <If condition={isUserUnAuthorizedForNamespace}>
+                <AuthorizationErrorMessage message={this.state.pageLevelError.message} />
+              </If>
               <If
-                condition={this.state.pageLevelError && this.state.pageLevelError.errorCode !== 404}
+                condition={
+                  !isUserUnAuthorizedForNamespace &&
+                  !isNamespaceNotFound &&
+                  this.state.pageLevelError
+                }
               >
                 <Page500
                   message={this.state.pageLevelError.message}
                   refreshFn={this.retrieveNamespace}
                 />
               </If>
-
+              {/**
+               * We show authorization failure message when the backend return empty
+               * namespace. This indicates the user has access to no namespace.
+               */}
               <If condition={!this.state.pageLevelError}>
-                {this.state.authorizationFailed ? <AuthorizationErrorMessage /> : container}
+                {this.state.authorizationFailed ? (
+                  <AuthorizationErrorMessage message={this.state.pageLevelError.message} />
```

Comment:
Long-term, should we show the message from the back end? Will it be localized? (No need to change for now)

Reply:
Yeah as of now we don't have the message from backend. One possible way to do this would be to have a health API from backend that deterministically say that the backend is down and if not can send a message that UI can use. But as far as I can see the message will be from UI as it needs to have call-to-actions as well as other context for the user.